tiledb/sm/misc/test/unit_mgc_dict.cc

```diff
@@ -471,35 +453,10 @@ int embedded_vs_external_identifications() {
     return 0;
   };
 
-  if (auto rval = proc_list(file_data_sizes1, true); rval != 0)
-    return rval;
-  if (auto rval = proc_list(file_data_sizes2, true); rval != 0)
-    return rval;
-
-  if (auto rval = proc_list(file_data_sizes1, false); rval != 0)
-    return rval;
-  if (auto rval = proc_list(file_data_sizes2, false); rval != 0)
-    return rval;
-
-  if (errcnt) {
-    fprintf(
-        stderr, "%d mismatch errors magic embedded vs external data\n", errcnt);
-  }
-
-  return errcnt ? 99 : 0;
-}
-
-int main() {
-  int failures = 0;
+  REQUIRE(proc_list(file_data_sizes1, true) == 0);
+  REQUIRE(proc_list(file_data_sizes2, true) == 0);
+  REQUIRE(proc_list(file_data_sizes1, false) == 0);
+  REQUIRE(proc_list(file_data_sizes2, false) == 0);
```

Comment:
These tests look like they're independent and should be in separate `SECTION` at least, if not separate `TEST_CASE`.

Reply:
I'm actually thinking of removing all but the first section. The two file data lists have the same content but in different order, and testing both with and without reusing the `magic_t` is not very worthwhile IMO.

Reply:
After offline feedback I removed all but the first case, and further simplified the test.